phases/17-infrastructure-and-production/21-ab-testing-llm-features/code/main.py

```
@@ -0,0 +1,92 @@
+    p_treat = p_baseline * (1 + lift)
+    z_alpha = 1.96
+    z_beta = 0.84
+    p_bar = (p_baseline + p_treat) / 2
+    num = (z_alpha * math.sqrt(2 * p_bar * (1 - p_bar)) +
+           z_beta * math.sqrt(p_baseline * (1 - p_baseline) + p_treat * (1 - p_treat))) ** 2
+    den = (p_treat - p_baseline) ** 2
+    return int(num / den)
+
+
+def simulate(p_a: float, p_b: float, seed: int = 7, max_n: int = 50_000) -> dict:
+    rng = random.Random(seed)
```

Ruff (0.15.11)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)